staging/src/k8s.io/client-go/transport/cache.go

```diff
@@ -60,7 +60,11 @@ func (c *tlsTransportCache) get(config *Config) (http.RoundTripper, error) {
 	}
 	// The options didn't require a custom TLS config
 	if tlsConfig == nil {
-		return http.DefaultTransport, nil
+		tr := http.DefaultTransport
+		if defaultTransport, ok := tr.(*http.Transport); ok {
+			defaultTransport.DisableCompression = config.DisableCompression
```

Comment:
maybe instead of using `http.DefaulTransport` here it makes more sense to create a "Default Client" (copy the fields of http.DefaultTransport into a new `*http.Transport`)? then apply `config.DisableCompression`